src/formatDate.js

```diff
@@ -50,7 +50,56 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  // Розділити рядок з датою на частини відповідно до старого формату
+  const dateParts = date.split(fromFormat[fromFormat.length - 1]);
+  const separator = `${toFormat[toFormat.length - 1]}`;
+  // Створити об'єкт зі значеннями дати відповідно до старого формату
+  const oldDateObj = {};
+  const pushToArr = (elem) => newDate.push(elem);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    oldDateObj[fromFormat[i]] = dateParts[i];
+  }
+
+  // Створити новий рядок з датою відповідно до нового формату
+  const newDate = [];
+
+  //! Не робив окрему перевірку на YYYY, YY
+  //!    - томущо всерівно прийшлося запускати цикл і коду в два рази більше.
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    switch (toFormat[i]) {
+      case 'YYYY':
+        if (oldDateObj['YYYY']) {
+          pushToArr(oldDateObj['YYYY']);
+        } else {
+          pushToArr(oldDateObj['YY'] < 30
+            ? `20${oldDateObj['YYYY'] || oldDateObj['YY']}`
+            : `19${oldDateObj['YYYY'] || oldDateObj['YY']}`);
+        }
+        break;
+
+      case 'YY':
+        if (oldDateObj['YYYY']) {
+          pushToArr(oldDateObj['YYYY'].slice(2));
+        } else {
+          pushToArr(oldDateObj['YY']);
+        }
+        break;
+
+      case 'MM':
+        pushToArr(oldDateObj['MM'].toString().padStart(2, '0'));
+        break;
+
+      case 'DD':
+        pushToArr(oldDateObj['DD'].toString().padStart(2, '0'));
```

Comment:
function `pushToArr` looks redundant to me, you can also create a variable for temporary result and push value to the array after switch/case